pkg/xds/topology/outbound.go

```diff
@@ -199,6 +199,10 @@ func buildExternalServiceEndpoint(externalService *mesh_core.ExternalServiceReso
 			mesh.GetMeta().GetName(), loader),
 		AllowRenegotiation: externalService.Spec.GetNetworking().GetTls().GetAllowRenegotiation().GetValue(),
 	}
+	if sni := externalService.Spec.GetNetworking().GetTls().GetSni(); sni != nil {
+		value := sni.GetValue()
+		es.SNI = &value
+	}
```

Comment:
```suggestion
es.SNI = externalService.Spec.GetNetworking().GetTls().GetSni().GetValue()
```

The protobuf wrappers take care of all the nil checks, and you'll end up with a value of "" if any intermediate message is `nil`.